### DISCUSSION TITLE
Function that wraps raw text as well as link

### DISCUSSION THREAD

**Discussion by CDucloux** [typst/typst#3834] [Q&A] [answered]
Hi there,

I'm trying to recreate my CV from `LaTeX` to `Typst`. I struggle to do something that I thought would be simple. 

I created a function called `website_link` which allows me to create a stylized *"sort of"* code block to link to my website, with a **FontAwesome** External Link Icon. Here's the function :

```typst
#let website_link(url, fill: luma(90%), size: 10pt) = {
  set text(black, size: size)
  block(
    text(
      link(url.dest) +
      h(0.15cm)
      + fa-external-link(size: (size - 1pt))
    ),
    fill: fill,
    inset: 0.5em,
    spacing: 0.75em
  )
}
```

> [!CAUTION]
> I would like to display the link as **raw text** but whenever I try to combine the `link` and `raw` functions I either loose the link capability or the raw formatting.

***

Typst version : 0.11.0

**Comment by Mc-Zen** [comment, marked as answer]
Hi there, 

the `link()` function accepts two arguments, one for the link string and one content argument that is displayed instead of the link text (if given). Does `link(url.dest, raw(url.dest))` do what you are looking for?

**Reply by CDucloux** [reply]
Indeed, it was what I was looking for. 

I used `set text(black, size: size, font: "Liberation Mono")` inside my `website_link` function to achieve the same thing.
